chain/bitcoind_conn.go

```diff
@@ -469,10 +469,16 @@ func (c *BitcoindConn) GetBlock(hash *chainhash.Hash) (*wire.MsgBlock, error) {
 		return nil, err
 	}
 
+	// cancelChan is needed in case a block request with the same hash is
+	// already registered and fails via the returned errChan. Because we
+	// don't register a new request in this case this cancelChan is used
+	// to signal the failure of the dependant block request.
+	cancelChan := make(chan error, 1)
```

Comment:
👍 

So just for my understanding, the issue before was that:
   * `N` goroutines try to fetch same block hash `H`
   * goroutine `N_0` gets there first
   * goroutines `N_1` to `N_i` block and wait for the response 
   * the query for goroutine `N_0` times out, that one exits
   * the others just sit there, and basically end up blocking forver
   * `synced_to_chain: false`

Reply:
Exactly this could happen but was acutally not observed by me in the wild but it defintely could happen if two requests for the same block come in shortly after another.

What I observed was that because we did not delete:

```
delete(d.blocksQueried, blockHash)
```
when we failed the request due to a timeout, we would never register a new request although there was no request going on in the background so we would wait forever for this new `getblock` calls.